test/github_pullrequest_concurrency_test.go

```diff
@@ -49,6 +56,40 @@ func TestGithubSecondPullRequestConcurrency1by1WithError(t *testing.T) {
 	})
 }
 
+func TestGithubGlobalRepoConcurrencyLimit(t *testing.T) {
+	ctx := context.Background()
+	label := "Github PullRequest Concurrent Two at a Time Set by Global Repo"
+	// create global repo
+	ctx, globalNS, runcnx, err := createGlobalRepo(ctx)
+	assert.NilError(t, err)
+	defer (func() {
+		err = cleanUpGlobalRepo(runcnx, globalNS)
+		assert.NilError(t, err)
+	})()
+
+	numberOfPipelineRuns := 10
+	// in this test we don't want to have `concurrency_limit` in local repo because
+	// global repository's `concurrency_limit` is being tested here.
+	maxNumberOfConcurrentPipelineRuns := -1
+	testGithubConcurrency(ctx, t, maxNumberOfConcurrentPipelineRuns, numberOfPipelineRuns, label, false, map[string]string{})
+}
+
+func TestGithubGlobalAndLocalRepoConcurrencyLimit(t *testing.T) {
+	ctx := context.Background()
+	label := "Github PullRequest Concurrent Three at a Time Set by Local Repo"
+	// create global repo
+	ctx, globalNS, runcnx, err := createGlobalRepo(ctx)
+	assert.NilError(t, err)
+	defer (func() {
+		err = cleanUpGlobalRepo(runcnx, globalNS)
+		assert.NilError(t, err)
+	})()
+
+	numberOfPipelineRuns := 10
+	maxNumberOfConcurrentPipelineRuns := 3
+	testGithubConcurrency(ctx, t, maxNumberOfConcurrentPipelineRuns, numberOfPipelineRuns, label, false, map[string]string{})
```

Comment:
Can you move this to different function and call that function from this test and TestGithubGlobalRepoConcurrencyLimit test to avoid duplication

Something like below

```
func TestGithubGlobalRepoConcurrencyLimit(t *testing.T) {
         testGlobalRepoConcurrency(t, "Github PullRequest Concurrent Two at a Time Set by Global Repo", -1)
}

func TestGithubGlobalAndLocalRepoConcurrencyLimit(t *testing.T) {
         testGlobalRepoConcurrency(t, "Github PullRequest Concurrent Three at a Time Set by Local Repo", -3)
}

func testGlobalRepoConcurrency(t *testing.T, label string, maxConcurrentPipelineRunsLocalRepo int) {
        ctx := context.Background()
	// create global repo
	ctx, globalNS, runcnx, err := createGlobalRepo(ctx)
	assert.NilError(t, err)
	defer (func() {
		err = cleanUpGlobalRepo(runcnx, globalNS)
		assert.NilError(t, err)
	})()

	numberOfPipelineRuns := 10
	maxNumberOfConcurrentPipelineRuns := 3
	testGithubConcurrency(ctx, t, maxConcurrentPipelineRunsLocalRepo, numberOfPipelineRuns, label, false, map[string]string{})
}
```